src/main/java/tdl/s3/upload/FileUploadingService.java

```diff
@@ -1,76 +1,53 @@
 package tdl.s3.upload;
 
 import com.amazonaws.services.s3.AmazonS3;
-import com.amazonaws.services.s3.model.ListMultipartUploadsRequest;
 import com.amazonaws.services.s3.model.MultipartUpload;
-import com.amazonaws.services.s3.model.MultipartUploadListing;
 
 import java.io.File;
-import java.util.Collection;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
+import tdl.s3.helpers.ExistingMultipartUploadFinder;
+import tdl.s3.sync.SyncProgressListener;
 
 public class FileUploadingService {
+
     private final AmazonS3 amazonS3;
+
     private final String bucket;
-    private String prefix;
+
+    private final String prefix;
+
+    private SyncProgressListener listener;
 
     public FileUploadingService(AmazonS3 amazonS3, String bucket, String prefix) {
         this.amazonS3 = amazonS3;
         this.bucket = bucket;
-	    this.prefix = prefix;
+        this.prefix = prefix;
+    }
+
+    public void setListener(SyncProgressListener listener) {
```

Comment:
To make sure we always have the right value for this listener, I would make it a mandatory constructor parameter.